lib/exercism/team_membership_invite.rb

```diff
@@ -0,0 +1,27 @@
+class TeamMembershipInvite < ActiveRecord::Base
+  belongs_to :team
+  belongs_to :user
+  belongs_to :inviter, class_name: 'User', foreign_key: :inviter_id
+
+  scope :for_team, -> (team_id) { where(team_id: team_id) }
+  scope :refused, -> { where(refused: true) }
+  scope :pending, -> { where(refused: false) }
+
+  def accept!
+    ActiveRecord::Base.transaction do
+      TeamMembership.create!(
+        team_id: team_id,
+        user_id: user_id,
+        inviter_id: inviter_id,
+        confirmed: true
+      )
+
+      destroy
+    end
+  end
+
+  def refuse!
+    self.refused = true
+    save
+  end
```

Comment:
What do we do with refused memberships later? Can they go back and change their mind to join the team? Or do we stop showing them?

This isn't something we need to necessarily change or decide now, I'm just curious what we should do.